onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -725,6 +726,13 @@ class PlannerImpl {
       ProcessDef(index, graph_viewer_.GetNodeArg(pair.first));
     }
 
+    // Check if should prefer the destination node EP.
+    auto UseEpSpecificAllocator = [](const std::string& provider_type) {
+      return provider_type == kQnnExecutionProvider ||
+             provider_type == kVitisAIExecutionProvider ||
+             provider_type == kVSINPUExecutionProvider;
+    };
```

Comment:
nit: this won't work with fully plugin EPs in the future. can we infer we should use an EP specific allocator by checking the alignment value of the allocator for CPU memory and using the allocator with the highest alignment value?

Reply:
This is not about alignment only. Can be for other reasons.
E.g. Qnn qtp allocator when enabled is not about alignment.

Reply:
how will that work with a plugin EP where we can't hardcode the name to get this behavior? doesn't need to be addressed in this PR but we need to understand what is required for this setup to work with a plugin EP if it can't be inferred from properties of the EP's allocator.

Reply:
still wondering if we can infer this. assuming it only applies to CPU memory

* if MemType differs, prefer the non-default one
  *  e.g. QNN uses OrtDevice::MemType::QNN_HTP_SHARED
  * if both are not default, no preference.
* prefer allocator with higher alignment requirement

Reply:
I will have to learn more about plugin EP. 
Off the top of my head, we could come up with a boolean EP interface that would ask us to override.

Simply having a memory allocator that allocates CPU accessible memory would not work. E.g. `CUDAPinned` is CPU accessible, but we are not asked to do this currently (we could).
It also depends on the performance results.

Another idea is to always call into EP and see if it has a CPU accessible preference, we could come up with an `enum` similar to `InputOnCPU`, and the EP returns optional override, we could use it. I think like it best.

Reply:
Nice suggestions with regards to selecting from multiple inputs.
Device allocators typically have a high alignment of 4K (different memtype), we could give those priority.